indico/modules/events/registration/controllers/display.py

```diff
@@ -135,17 +135,31 @@ class RHRegistrationFormCheckEmail(RHRegistrationFormBase):
 
     def _process(self):
         email = request.args['email'].lower().strip()
-        if self.regform.get_registration(email=email):
-            return jsonify(conflict='email')
         user = get_user_by_email(email)
-        if user and self.regform.get_registration(user=user):
-            return jsonify(conflict='user')
-        elif user:
-            return jsonify(user=user.full_name, self=(user == session.user))
-        elif self.regform.require_user:
-            return jsonify(conflict='no-user')
+        update = request.args.get('update')
+
+        if update:
+            existing = self.regform.get_registration(uuid=update)
+            if existing.user != user:
+                return jsonify(conflict='email-not-user')
+            elif user:
+                return jsonify(user=user.full_name, self=(user == session.user))
+            elif self.regform.require_user:
```

Comment:
Do we really want to check `require_user` for modifications? I don't think it would be expected from the definition we have of it.